cmd/build/build.go

```diff
@@ -145,6 +144,22 @@ func runBuild(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	productName, err := cmd.Flags().GetString("product-name")
+	if err != nil {
+		return err
+	}
+
+	// Use product-name if provided, otherwise use name
+	if productName != "" {
+		name = productName
+	}
```

Comment:
This is a replacement for the existing `name` flag. This need to be mutually exclusive. Also deprecate the old flag and remove it from the usage / help notes. Keep the old flag for backwards-compatibility but make it explicit that it is deprecated and to use the new flag